tests/unit/test_download.py

```diff
@@ -54,6 +54,12 @@ def test_user_agent():
     PipSession().headers["User-Agent"].startswith("pip/%s" % pip.__version__)
 
 
+def test_user_agent_provisional_string(monkeypatch):
+    provisionalString = "provisional_string"
+    monkeypatch.setenv("PIP_USER_AGENT_PROVISIONAL_STRING", provisionalString)
+    provisionalString in PipSession().headers["User-Agent"]
```

Comment:
FYI, you need to add `assert` at the beginning of this line. Otherwise the test will always pass. (The test above is also missing an `assert`.)